src/ScaffoldInstallerPlugin.php

```diff
@@ -468,10 +468,26 @@ function($key) {
         // Add Redis service.
         if (file_exists('./.ddev/docker-compose.redis.yaml')) {
             $redisConfig = Yaml::parseFile('.ddev/docker-compose.redis.yaml');
-            $redisImage = explode(':',
-                $redisConfig['services']['redis']['image']);
+            $image = $redisConfig['services']['redis']['image'] ?? '';
+        
+            // Normalize image from possible environment syntax.
+            // Example: $image = '${REDIS_DOCKER_IMAGE:-redis:7}'.
+            if (preg_match('/^\$\{[^:}]+:-(.+)\}$/', $image, $matches)) {
+                $image = $matches[1];
+            }
+            // Example: $image = 'redis:${REDIS_TAG:-6-bullseye}'.
+            elseif (preg_match('/^([^:]+):\$\{[^:}]+:-(.+)\}$/', $image, $matches)) {
+                $image = "{$matches[1]}:{$matches[2]}";
+            }
+        
+            // Extract the version/tag from the image string.
+            // Example: $image = 'redis:6-bullseye' → $version = '6-bullseye'.
+            if (!preg_match('/:([a-zA-Z0-9._-]+)$/', $image, $versionMatch)) {
+                throw new \RuntimeException("Unable to extract Redis version from image: {$image}");
+            }
```

Comment:
Let's avoid code duplication and instead just call this new function.

`extractRedisImageVersion(string $image)`